client/my-sites/email/titan-new-mailbox/index.jsx

```diff
@@ -118,6 +134,16 @@ const TitanNewMailbox = ( {
 						/>
 					</FormLabel>
 					{ hasMailboxError && <FormInputValidation text={ mailboxError } isError /> }
+					{ backPath && (
+						<FormSettingExplanation>
+							{ translate( 'You still can modify your domain. {{a}}Go back{{/a}}.', {
+								components: {
+									// eslint-disable-next-line jsx-a11y/anchor-is-valid
+									a: <a href="#" onClick={ redirectAndMaybeRemoveDomainFromCart } />,
```

Comment:
I wonder if we really need to clean up the cart, as this will only work when the user clicks that particular link: if the user use the back button from their browser, the original domain will still be present in the cart.

Reply:
Maybe it's good to keep it to prevent unintentional buying and therefore refunds? At least one of the two actions is helping the user to do not to be charged unintentionally.